Add option to redact http headers (#26724)
cilium/cilium: Documentation/observability/visibility.rst

the following features and values for supported Layer 7 protocols:

* For HTTP: redacting URL query (GET) parameters (``--hubble-redact=http-url-query``)
* For HTTP: redacting all headers except of specific ones (``--hubble-redact=http-headers``).

>   Traceparent, Tracestate, Cache-Control, Content-Encoding, and Accept headers are not redacted.

Reviewer: ```suggestion
  The following headers are not redacted: <list here>
```

* For Kafka: redacting API key (``--hubble-redact=kafka-api-key``)

> 

Reviewer: ```suggestion
```
